### DISCUSSION TITLE
Roadmap - need your help!

### DISCUSSION THREAD

**Discussion by Okelm** [TheWidlarzGroup/react-native-video-player#186] [General]
Hi! 
Long story short: There are a couple of react native video player libraries in the space. At @TheWidlarzGroup we're trying to understand what is the state of the controls to understand your pain points and build the roadmap from there. 

Whether you're actively using this library in your project or not, but you do use some UI controls - please chip in and let me know what's missing!

**Comment by coofzilla** [comment]
One of our biggest issues is the inability to add custom overlays, buttons, additional functionality, etc, while in fullscreen. 

We're using https://www.npmjs.com/package/react-native-media-console with locker to kind of mitigate that; but, adding another layer of abstraction isn't ideal. 

Would definitely love if `react-native-video-layer` supported this out of the box.

**Reply by moskalakamil** [reply]
Would be great! True fullscreen includes default controls that can't be modified, but we could simulate fullscreen behavior and create custom controls instead

**Reply by RoyRao2333** [reply]
@moskalakamil 

Hi, may I ask what the current progress is? 😊

**Comment by Gautham495** [comment]
This is the player I made with rn-video and it is already in production for both audio and video content.

https://x.com/gautham_vijay_/status/1822346356574318868/video/1

I am not getting the time to open source it.

But this works on par with big apps.

**Reply by moskalakamil** [reply]
Looks amazing!

**Reply by guihendias** [reply]
Wow, it looks fantastic. I would love to use it! 

LMK if I can help you open source it!

**Reply by Gautham495** [reply]
Sure @guihendias.

It also has support to google cast and airplay.

Also has support for thumbnail scrubbing with mux.com's storyboard JSON feature.

It still needs some fine tuning but i will share the full code when i can.

**Comment by imohsinali** [comment]
is it work with bridgeless mode in react native as react native video does not support new bridgeless mode

**Reply by moskalakamil** [reply]
Hey, this library relies on react-native-video, so it shares the same issues as react-native-video
